symengine/basic.cpp

```diff
@@ -24,6 +43,20 @@ std::string Basic::__str__() const
     return str(*this);
 }
 
+std::string Basic::dumps() const {
+    std::ostringstream oss;
+    cereal::BinaryOutputArchive{oss}(this->rcp_from_this());
```

Comment:
Is `BinaryOutputArchive` portable?

Reply:
Thanks, no it wasn't endianness is handled by PortableBinaryOutputArchive. Will change this.